aptos-move/framework/aptos-framework/sources/fungible_asset/managed_fungible_metadata.move

```diff
@@ -0,0 +1,294 @@
+/// This module provides an addtional ready-to-use solution on top of `FungibleAssetMetadata` that manages the refs of
+/// mint, burn and transfer for the creator in a straightforward scheme. It offers creators to destory any refs in an
+/// on-demand manner too.
+module aptos_framework::managed_fungible_metadata {
+    use aptos_framework::fungible_asset::{Self, MintRef, TransferRef, BurnRef, FungibleAsset};
+    use aptos_framework::fungible_store;
+    use aptos_framework::object::{Self, Object, ConstructorRef};
+    use std::error;
+    use std::option::{Self, Option};
+    use std::signer;
+    use std::string::String;
+
+    /// MintRef existence error.
+    const EMINT_REF: u64 = 1;
+    /// TransferRef existence error.
+    const ETRANSFER_REF: u64 = 2;
+    /// BurnRef existence error.
+    const EBURN_REF: u64 = 3;
+    /// Not the owner.
+    const ENOT_OWNER: u64 = 4;
+    /// Refs existence errors.
+    const EMANAGED_FUNGIBLE_ASSET_REFS: u64 = 5;
+
+    #[resource_group_member(group = aptos_framework::object::ObjectGroup)]
+    /// Hold refs to control the minting, transfer and burning of fungible assets.
+    struct ManagingRefs has key {
+        mint: Option<MintRef>,
+        transfer: Option<TransferRef>,
+        burn: Option<BurnRef>,
+    }
+
+    /// Initialize metadata object and store the refs.
+    public fun init_managing_refs(
+        constructor_ref: &ConstructorRef,
+        maximum_supply: u64,
+        name: String,
+        symbol: String,
+        decimals: u8
+    ) {
+        let (mint_ref, transfer_ref, burn_ref) = fungible_asset::init_metadata(
+            constructor_ref,
+            maximum_supply,
+            name,
+            symbol,
+            decimals
+        );
+        let metadata_object_signer = object::generate_signer(constructor_ref);
+        move_to(
+            &metadata_object_signer,
+            ManagingRefs {
+                mint: option::some(mint_ref), transfer: option::some(transfer_ref), burn: option::some(burn_ref)
+            }
+        )
+    }
+
+    /// Mint as the owner of metadata object.
+    public fun mint<T: key>(
+        metadata_owner: &signer,
+        metadata: &Object<T>,
+        amount: u64,
+        to: address
+    ) acquires ManagingRefs {
+        assert_owner(metadata_owner, metadata);
+        let mint_ref = borrow_mint_from_refs(metadata);
+        let fa = fungible_asset::mint(mint_ref, amount);
+        fungible_store::deposit(fa, to);
```

Comment:
I don't think this module should know about fungible store

Reply:
That's quite different than I thought. Then basically we have to build another layer otherwise most users will reinvent the wheel. Agree?